vault/seal/seal_testing.go

```diff
@@ -1,89 +1,24 @@
 package seal
 
 import (
-	"context"
-
-	"github.com/hashicorp/vault/helper/xor"
-	"github.com/hashicorp/vault/sdk/physical"
+	"github.com/hashicorp/go-hclog"
+	wrapping "github.com/hashicorp/go-kms-wrapping"
 )
 
-type TestSeal struct {
-	Type   string
-	secret []byte
-	keyId  string
-}
-
-var _ Access = (*TestSeal)(nil)
-
-func NewTestSeal(secret []byte) *TestSeal {
-	return &TestSeal{
-		Type:   Test,
-		secret: secret,
-		keyId:  "static-key",
-	}
-}
-
-func (s *TestSeal) Init(_ context.Context) error {
-	return nil
-}
-
-func (t *TestSeal) Finalize(_ context.Context) error {
-	return nil
-}
-
-func (t *TestSeal) SealType() string {
-	return t.Type
-}
-
-func (t *TestSeal) KeyID() string {
-	return t.keyId
-}
-
-func (t *TestSeal) SetKeyID(k string) {
-	t.keyId = k
+type TestSealOpts struct {
+	Logger     hclog.Logger
+	StoredKeys StoredKeysSupport
+	Secret     []byte
+	Name       string
 }
 
-func (t *TestSeal) Encrypt(_ context.Context, plaintext []byte) (*physical.EncryptedBlobInfo, error) {
-	ct, err := t.obscureBytes(plaintext)
-	if err != nil {
-		return nil, err
+func NewTestSeal(opts *TestSealOpts) *Access {
```

Comment:
I wonder if this name might create confusion with vault.NewTestSeal?

Reply:
Maybe, but it's also the name that's currently in use in this file. It could be changed later.